### ISSUE TITLE
Pluto occasionally gets stuck with very old package version in notebook

### ISSUE THREAD

**Issue by BioTurboNick** [fonsp/Pluto.jl#2275]
So, I'm not sure how well I'll be able to reproduce this. And I'm not sure if it's PlutoPkg or Pkg that's really the issue.

The issue seems to only occur with Transducers.jl, that I've seen.

I'll want to work on a method from within my direct dependency, so I'll load the dependencies it needs as `using` in separate cells (mainly StaticArrays and NearestNeighbors). But for some reason, this sometimes leads to a secondary dependency, Transducers.jl, being downgraded from v0.4.73 to v.0.1.0, which I can tell because previously working code throws a MethodError.

If I then add `using Transducers` to a cell, the cloud icon hover tells me it'll install v0.4.73, the latest one. But when run, it's v0.1.0. If I then instruct Pluto to update the packages again, it will then install v0.4.73 and things will work again.

If I see this happen again, what kind of information can I collect that may help?

**Comment by fonsp** [comment]
> Transducers.jl, being downgraded from v0.4.73 to v.0.1.0, which I can tell because previously working code throws a MethodError.

Does it throw the methoderror after restarting the notebook process, or directly after adding the new imports? Does Pluto recommend a restart?

> If I then add using Transducers to a cell, the cloud icon hover tells me it'll install v0.4.73, the latest one.

The version that Pluto tells here is not actually accurate 🤫(because of https://github.com/JuliaLang/Pkg.jl/issues/11), it just shows the latest version number. So this number is only for illustration, in the end, Pkg.jl decides.

If it happens again, save an HTML export, and try to retrace your steps in a new notebook to reproduce the issue.

I might have a similar issue: https://github.com/fonsp/package-loading-times/issues/2

**Comment by BioTurboNick** [comment]
You're right, it seems to be something with Pkg. I ran Pkg.up() on my global environment and it caused the downgrade there too. I'll open an issue over there.

**Comment by BioTurboNick** [comment]
Okay, so this is what was happening for me:

Running `update` will update each package, in order, as far as it can. If an earlier indirect dependency is updated first, that can silently constrain later packages.

If you then explicitly update the problem package, it will allow you to do so, likely downgrading that earlier package. But then if you run the general update again, the problem will recur because it again updates the earlier package.

So to resolve, you may need to add a compat entry to the dependency that keeps being downgraded, and/or there may be broken compat entries elsewhere in the ecosystem that need to be updated.

**Comment by BioTurboNick** [comment]
I think the only actionable suggestion here might be a way to see the versions of indirect dependencies installed, and/or a way to manipulate compat entries for the notebook itself?
